### PR TITLE
Allow dependson and dependsOn

### PR DESCRIPTION
**Pull request by zuzuleinen**
Add support for both `dependson` and `dependsOn` on the task level.

### CODE REVIEW

File: bobtask/task.go [hunk 1]
```diff
@@ -388,3 +390,41 @@ func keyExists(m map[string]interface{}, key string) bool {
...
+	var values struct {
+		Lowercase []string `yaml:"dependson"`
+		Camelcase []string `yaml:"dependsOn"`
+	}
+
+	err = value.Decode(&values)
+	errz.Fatal(err)
+
+	if len(values.Lowercase) > 0 && len(values.Camelcase) > 0 {
+		errz.Fatal(errors.New("both `dependson` and `dependsOn` exists for task"))
```
Comment (Equanox) [inline bobtask/task.go:406]: This shouldn't be a user error. Is it possible to annotate the error message with the actual task name?

**Review by Equanox — CHANGES_REQUESTED** [review]
Please also do for run tasks

File: bobtask/task.go (hunk 1 above)
Comment (zuzuleinen) [reply bobtask/task.go:406]: I tried to see if I can add the name, but the key(used for name) is 1-level higher and couldn't manage to get it

Comment (zuzuleinen) [reply bobtask/task.go:406]: line

for this value *yaml.Node will have .Line 25. maybe display the line?

Comment (Equanox) [reply bobtask/task.go:406]: If this is the best we can get.. Yes, please!

**Review by Equanox — APPROVED** [review]

**Pull request merged**